added the signup frontend

swostikpati opened the pull request:
also made the start commands work in windows: npm run start-w and npm run start-local-backend-w

swostikpati committed:
added the signup frontend

nguyenvince committed:
minor changes

nguyenvince committed:
minor stylistic changes

nguyenvince commented:
Good progress! A few minor UI improvements below.

### Signup form validation
```
            sx={{
              "& .MuiOutlinedInput-root": {
                borderColor:
                  password.length >= MINIMUM_PASSWORD_LENGTH && password === confirmPassword ? "green" : "",
                "&.Mui-focused fieldset": {
                  borderColor:
                    password.length >= MINIMUM_PASSWORD_LENGTH && password === confirmPassword ? "green" : "",
                },
              },
            }}
```

I see what you're trying to do here, but the effect is not very noticeable because there are a few missing actions. See the video and let me know if you can implement these changes: https://drive.google.com/file/d/12Yp2ndPwKtioBnce6pRuv7Z3o7RqahTB

I made some stylistic changes as well.

nguyenvince commented:
also, you can add email validation for the `signup`, it's okay because here we're taking email exclusively.

you can refer to the `EmailsInput.jsx` to see how the validation was done. you can create a helper function that can be shared between the signup and this one then, don't duplicate codes. the helper function can be stored somewhere in `Utils`

swostikpati committed:
changed password validation

swostikpati committed:
added email validation and made minor changes to password validation

swostikpati commented:
Made all the required changes.

For the email validation, I only display green and not red outline as it isn't an intuitive thing for an user to encounter a red outline in the textbox until they finish typing their email. However they will be greeted with a green outline as soon as they type in a proper email. Also if they try to submit the form without a valid email, the validation function would kick in and display a message.

nguyenvince committed:
setCurrentPage for SignUp

nguyenvince commented:
<img width="362" alt="Screenshot 2024-11-27 at 14 46 05" src="https://github.com/user-attachments/assets/17a78612-add8-441b-b3b6-c2578842709e">

great! I think to make it perfect, we can follow the amz example in the screenshot below where each textfield is accompanied by one explainer line to make it even clearer what the reqs are, instead of just color. color is nice, but I think text will be the perfect explanation. feel free to copy what amz did in terms of focusing/unfocusing the fields and when the explainer text appears/validates

nguyenvince committed:
removed redirect between login and signup

nguyenvince committed:
harmonize UI for school and personal account

- added instructions for login/signup for schools and nyuad
- removed nyuadDashboardNavLink
- School Login --> login / Signup
- show Alert button for NYUAD
- route to `/dashboard/nyuad` if no valid allowedSchools is found for user

swostikpati committed:
added text under input fields

swostikpati commented:
added text below the password and confirm password text fields like the amazon sign up. Again didn't do it for email cause didn't want to overwhelm the user with a red outline and email format is obvious. It would still give a popup toast error in case they made an error with the email while submitting

nguyenvince committed:
removed hardcoding, added placeholder for pwd

nguyenvince commented:
sounds good

I made some minor adjustments

swostikpati committed:
added dialog box for email verification

swostikpati commented:
dialog box added!

swostikpati committed:
added verify route in frontend

swostikpati commented:
added the verify route. Waiting on a valid token to test

swostikpati committed:
fixed some bugs and tested verify route

nguyenvince committed:
return to "/" in Login and Signup if already logged in

nguyenvince committed:
signup and google oauth integration

- added npm run build-local-backend
- handled login/signup/google oauth
- handled error messages, displayed them with SnackbarNotification
- refactored `fetchDataFromURL`

nguyenvince committed:
implemented Google OAuth as popup instead

nguyenvince committed:
bug fixed

nguyenvince committed:
added back status

nguyenvince committed:
expire promo banners

nguyenvince committed:
added ProjectReservedArea

- requires login, even for public school `nyuad`, for TV screen, dataset, and especially ALERT (requires a verified email address for public school `nyuad`)
- doesn't allow to open these Dialogs if not logged in
- created Login (password) popup, in addition to existing GoogleOAuth Login popup
- misc: fixed unique key issues

nguyenvince committed:
transitioned to using notistack instead of MUI default Snackbar

nguyenvince committed:
Update Verify.jsx

nguyenvince committed:
fixed <a> not showing up
